bandwidth-file-spec.txt

```diff
@@ -479,95 +621,260 @@
+
+      This KeyValue was added in version 1.4.0 of this specification.
+
+    "recent_measurements_excluded_few_count=" Int
+
+      [Zero or one time.]
+
+      The number of successful measurements for this relay that were ignored
+      because the relay did not have enough successful measurements (fewer
+      than 2, by default).
+
+      (See the note in section 1.4, version 1.4.0, about excluded relays.)
+
+      This KeyValue was added in version 1.4.0 of this specification.
+
+    "vote=" bool
+
+      [Zero or one time.]
+
+      If the value is 0, Tor directory authorities SHOULD ignore the relay's
+      entry in the bandwidth file. They SHOULD vote for the relay the same
+      way they would vote for a relay that is not present in the file.
+
+      This MAY be the case when this relay was not successfully measured but
+      the is included in the Bandwidth File, to diagnose why they were not
+      measured.
+
+      If the value is 1 or the KeyValue is not present, Tor directory
+      authorities MUST use the relay's bw value in any votes for that relay.
+
+      Because Tor versions 0.3.4.11, 0.3.5.8, 0.4.0.2-alpha, and earlier
+      ignore this KeyValue, generator implementations MUST set "bw=1"
+      for a relay when Tor directory authorities SHOULD ignore its relay
+      line. Using the minimum bw value allows authorities that do not
+      understand "vote=0" or "unmeasured=1" to produce sensible votes
+      for unmeasured relays.
+
+      This KeyValue was added in version 1.4.0 of this specification.
+
+    "unmeasured=" bool
+
+      [Zero or one time.]
+
+      If the value is 1, this relay was not successfully measured and
+      Tor bandwidth authorities MAY NOT vote on this relay.
+      (Current Tor versions do not change their behaviour based on
+      the "unmeasured" key.)
+
+      If the value is 0 or the KeyValue is not present, this relay
+      was successfully measured.
+
+      Implementations MUST also set "bw=1" for this relay when it was not
+      successfully measured. (See the explanation under "vote" for more
+      details.)
+
+      This KeyValue was added in version 1.4.0 of this specification.
+
+    "under_min_report=" bool
+
+      [Zero or one time.]
+
+      If the value is 1, there are not enough eligible relays in the
+      bandwidth file, and Tor bandwidth authorities MAY NOT vote on this
+      relay. (Current Tor versions do not change their behaviour based on
+      the "under_min_report" key.)
+
+      If the value is 0 or the KeyValue is not present, there are enough
+      relays in the bandwidth file.
+
+      When there are not enough eligible relays in the bandwidth file,
+      implementations MUST also set "bw=1" for all relays. (See the
+      explanation under "vote" for more details.) The actual measured
+      bandwidth MAY be included as "real_bw=".
+
+      This KeyValue was added in version 1.4.0 of this specification.
+
+    "real_bw=" Int
```

Comment:
Hmm, we're not calculating the bandwidth for the relays excluded (https://github.com/torproject/sbws/blob/6070d36022dc2130518dd0c68332166b2bf76c73/sbws/lib/v3bwfile.py#L618), though the case here is different.
Since Tor is not going to use it for now, do we really want to include this in this version?.

Reply:
See my response in:
https://trac.torproject.org/projects/tor/ticket/29853#comment:7

I want to remove real_bw, and use the measured/filtered/scaled bandwidth for relays exluded by under_min_report.

Reply:
I rewrote this part in c8cf8db.